pkg/controller/serviceaccount/tokens_controller.go

```diff
@@ -707,6 +707,29 @@ func (e *TokensController) listTokenSecrets(serviceAccount *v1.ServiceAccount) (
 	return items, nil
 }
 
+func (e *TokensController) updateServiceAccount(ctx context.Context, namespace string, sa *v1.ServiceAccount) error {
+	// Remove the secret if not exist.
+	if len(sa.Secrets) != 0 {
+		var refs []v1.ObjectReference
+		for _, ref := range sa.Secrets {
+			_, err := e.client.CoreV1().Secrets(namespace).Get(ctx, ref.Name, metav1.GetOptions{})
```

Comment:
doing N live gets on every service account update is a significant change… do you have details about the bug this is trying to fix? A test demonstrating that bug would be a helpful starting point

Reply:
#98415 it to reproduce this bug 
``` shell
while true ; do
kubectl delete secrets --all
done
```

I triggered this problem while upgrading k8s and needed to recreate the SA to fix it